server/src/main/java/io/unitycatalog/server/persist/SchemaRepository.java

```diff
@@ -176,22 +185,73 @@ public SchemaInfo updateSchema(String fullName, UpdateSchema updateSchema) {
         }
     }
 
-    public void deleteSchema(String fullName) {
+    public void deleteSchema(String fullName, boolean force) {
         try (Session session = sessionFactory.openSession()) {
+            String[] namespace = fullName.split("\\.");
+            if (namespace.length != 2) {
+                throw new BaseException(ErrorCode.INVALID_ARGUMENT,
+                        "Invalid schema name: " + fullName);
+            }
+            CatalogInfoDAO catalog = catalogRepository.getCatalogDAO(session, namespace[0]);
+            if (catalog == null) {
+                throw new BaseException(ErrorCode.NOT_FOUND,
+                        "Catalog not found: " + namespace[0]);
+            }
             Transaction tx = session.beginTransaction();
             try {
-                SchemaInfoDAO schemaInfo = getSchemaDAO(session, fullName);
-                if (schemaInfo != null) {
-                    session.remove(schemaInfo);
-                    tx.commit();
-                } else {
-                    throw new BaseException(ErrorCode.NOT_FOUND,
-                            "Schema not found: " + fullName);
-                }
+                deleteSchema(session, catalog.getId(), namespace[1] ,force);
+                tx.commit();
             } catch (Exception e) {
                 tx.rollback();
                 throw e;
             }
         }
     }
+
+    public void deleteSchema(Session session, UUID catalogId, String schemaName, boolean force) {
+        SchemaInfoDAO schemaInfo = getSchemaDAO(session, catalogId, schemaName);
+        if (schemaInfo != null) {
+
+            // handle tables
+            List<TableInfoDAO> tables = tableRepository.
```

Comment:
It's better to unify the handling for different entity types so that we don't miss new entities types.

Reply:
There's no field currently in the db schema to indicate if an entity is level-3 entity (for e.g. table, function, volume). Also there is one different singleton instance of `repository` associated with each type(i.e.`TableRepository`, `FunctionRepository`, `VolumeRepository` are all different classes and we maintain singletons of each). Given this structure, I am not sure what would be a good way to unify the handling of all entity types.